auslib/db.py

```diff
@@ -403,14 +403,13 @@ def _prepareInsert(self, trans, changed_by, **columns):
         if self.versioned:
             data['data_version'] = 1
         query = self._insertStatement(**data)
-
-        if self.onInsert:
-            self.onInsert(self, "INSERT", changed_by, query, trans)
-
         ret = trans.execute(query)
         if self.history:
             for q in self.history.forInsert(ret.inserted_primary_key, data, changed_by):
                 trans.execute(q)
+        if self.onInsert:
+            pk_column = self.t.primary_key.columns.keys()[0]
```

Comment:
Right now, I have no access to sqlalchemy doc, but doing `self.t.primary_key.columns.keys()[0]`  will get part of a possible composable key or a composable key as whole?

Reply:
I believe `.keys()` returns a list of primary keys.. So if the table has only 1 pk then the `[0]` indexing gets it.

Reply:
Yep, some tables can has more than one primary key column, so getting zero index will get only one of them.

Reply:
Do we want all primary keys fields in the email?

Reply:
Makes sense for me, once the function is generic. What you think?

@mozbhearsum any thought about that?

Reply:
Yeah, I think that makes sense.

Reply:
I think this is probably why the permissions e-mail broke -- you need to account for the fact that `ret.inserted_primary_key` may contain multiple columns.